Replace pattern matching with if/then/else in Chapter 4 examples. (#272)
purescript-contrib/purescript-book: text/chapter4.md

```haskell
fib :: Int -> Int
fib n =
  if n == 0 || n == 1 then
    1
  else
    fib (n - 1) + fib (n - 2)
```

Again, this problem is solved by considering the solutions to subproblems. In this case, there are two subproblems, corresponding to the expressions `fib (n - 1)` and `fib (n - 2)`. When these two subproblems are solved, we assemble the result by adding the partial results.

> (Note that, while the above examples of `fact` and `fib` work as intended, a more idiomatic implementation would handle the base cases with pattern matching, as discussed in a later chapter.)

Reviewer: ```suggestion
Note that, while the above examples of `fact` and `fib` work as intended, a more idiomatic implementation would use pattern matching instead of `if`/`then`/`else`. Pattern matching techniques are discussed in a later chapter.
```

Don't know if "base cases" is the best term here. Technically _all cases_ would be handled by pattern matching, since `fib n` and `fact n` are not the base cases.
Also wanted to emphasize what pattern matching replaces.

Author: Yes, your rewrite sounds reasonable to me.
